analyzers/tests/SonarAnalyzer.UnitTest/Helpers/MethodParameterLookupTest.cs

```diff
@@ -25,287 +25,374 @@
+        argumentEx.Message.Should().StartWith("argument must be of type Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentSyntax");
+    }
 
-            var argumentEx = Assert.ThrowsException<ArgumentException>(() =>
-                lookupThrow.TryGetSymbol(VBCodeAnalysis.SyntaxFactory.StringLiteralExpression(VBCodeAnalysis.SyntaxFactory.StringLiteralToken(string.Empty, string.Empty)), out var parameter));
-            argumentEx.Message.Should().StartWith("argument must be of type Microsoft.CodeAnalysis.VisualBasic.Syntax.ArgumentSyntax");
-        }
+    [TestMethod]
+    public void TestMethodParameterLookup_VB_ThrowsException()
+    {
+        var c = new VisualBasicInspection(SourceVB);
+        var lookupThrow = c.CreateLookup(1, "DoSomething");
 
-        private abstract class InspectionBase<TArgumentSyntax, TInvocationSyntax>
-            where TArgumentSyntax : SyntaxNode
-            where TInvocationSyntax : SyntaxNode
-        {
-            public SnippetCompiler Compiler { get; protected set; }
-            public TInvocationSyntax[] MainInvocations { get; protected set; }
-            public TArgumentSyntax SpecialArgument { get; private set; }
-            public IParameterSymbol SpecialParameter { get; private set; }
+        var invalidOperationEx = Assert.ThrowsException<InvalidOperationException>(() => lookupThrow.TryGetNonParamsSyntax(lookupThrow.MethodSymbol.Parameters.Single(), out var argument));
+        invalidOperationEx.Message.Should().Be("Sequence contains more than one element");
 
-            public abstract TInvocationSyntax[] FindInvocationsIn(string name);
-            public abstract object ExtractArgumentValue(TArgumentSyntax argumentSyntax);
-            public abstract TArgumentSyntax[] GetArguments(TInvocationSyntax invocation);
-            public abstract MethodParameterLookupBase<TArgumentSyntax> CreateLookup(TInvocationSyntax invocation, IMethodSymbol method);
+        var argumentEx = Assert.ThrowsException<ArgumentException>(() =>
+            lookupThrow.TryGetSymbol(VBCodeAnalysis.SyntaxFactory.StringLiteralExpression(VBCodeAnalysis.SyntaxFactory.StringLiteralToken(string.Empty, string.Empty)), out var parameter));
+        argumentEx.Message.Should().StartWith("argument must be of type Microsoft.CodeAnalysis.VisualBasic.Syntax.ArgumentSyntax");
+    }
 
-            protected InspectionBase(string source, AnalyzerLanguage language)
+    [TestMethod]
+    public void TestMethodParameterLookup_CS_MultipleCandidates()
+    {
+        var source = """
+            namespace Test
             {
-                Compiler = new SnippetCompiler(source, false, language);
-                MainInvocations = FindInvocationsIn("Main");
-            }
+                class TestClass
+                {
+                    void Main()
+                    {
+                        dynamic d = 42;
+                        AmbiguousCall(d);
+                    }
```

Comment:
This saves some lines:
```suggestion
                    void Main(dynamic d) =>
                        AmbiguousCall(d);
```